sudoku.py

```diff
@@ -164,11 +149,12 @@ def main():
                 # selected.
                 bqm.fix_variable(get_label(row, col, value), 1)
 
-    # Solve BQM
+    return bqm
+
+def solve_sudoku(bqm, matrix):
```

Comment:
I'd make it a pure (no side-effects) function, and return the result matrix instead of doing the inplace update.